client/state/posts/actions.js

```diff
@@ -62,11 +63,20 @@ export function receivePosts( posts ) {
  * @return {Function}        Action thunk
  */
 export function requestSitePosts( siteId, query = {} ) {
+	const q = {
+		...DEFAULT_POST_QUERY,
+		...query
+	};
+
+	if ( q.status === 'draft,pending' ) {
```

Comment:
I'm not sure this is the right place to modify query args. I agree with @aduth this should be handled on a component level. I'd expect for this action creator to be flexible and work with whatever query args I pass to it; I wouldn't expect that it would modify them for me.

One approach we can go with is to use a utility similar to `mapPostStatus`, or it would be even better if we have one centralized (and easily testable) utility function that takes care of these modifications for us. But we'd call it in the component, and not in the action creator.